test/unit/cli_test.py

```diff
@@ -0,0 +1,186 @@
+                        )
+                        """
+            ),
+            [
+                Validation.Error(
+                    message="error-codes only can contain constant values, details: <class 'ast.Name'>",
+                    file="<Unknown>",
+                    line_number=7,
+                )
+            ],
+        ),
+        (
+            cleandoc(
+                """
+                        from exasol import error
+                        from exasol.error import Parameter
+        
+                        var = input("description: ") 
+        
+                        error1 = error.ExaError(
+                            "E-TEST-1",
+                            "this is an error",
+                            [var],
+                            {"param": Parameter("value", "description")},
+                        )
+                        """
+            ),
+            [
+                Validation.Error(
+                    message="mitigations only can contain constant values, details: <class 'ast.Name'>",
+                    file="<Unknown>",
+                    line_number=9,
+                )
+            ],
+        ),
+        (
+            cleandoc(
+                """
+                        from exasol import error
+                        from exasol.error import Parameter
+        
+                        var = input("description: ") 
+        
+                        error1 = error.ExaError(
+                            "E-TEST-1",
+                            "this is an error",
+                            var,
+                            {"param": Parameter("value", "description")},
+                        )
+                        """
+            ),
+            [
+                Validation.Error(
+                    message="mitigations only can contain constant values, details: <class 'ast.Name'>",
+                    file="<Unknown>",
+                    line_number=9,
+                )
+            ],
+        ),
+    ],
+)
+def test_ErrorCollector_errors(src, expected):
+    root_node = ast.parse(src)
+    collector = ErrorCollector(root_node)
+    collector.collect()
+    assert expected == collector.errors
+
+
+@pytest.mark.parametrize(
+    ["src", "expected"],
+    [
+        (
+            cleandoc(
+                """
+                        from exasol import error
+                        from exasol.error import Parameter
+        
+                        var = input("description: ") 
+        
+                        error1 = error.ExaError(
+                            "E-TEST-1",
+                            "this is an error",
+                            ["no mitigation available"],
+                            {"param": Parameter("value", var)},
+                        )
+                        """
+            ),
+            [],
+        ),
+    ],
+)
+def test_ErrorCollector_warnings(src, expected):
```

Comment:
do we have no warnings, yet, that you only test that there were none

Reply:
https://github.com/exasol/error-reporting-python/issues/23